Makefile

```diff
@@ -12,130 +12,222 @@
-
-IMAGE   ?= amazon/amazon-k8s-cni
-VERSION ?= $(shell git describe --tags --always --dirty)
+.PHONY: all dist check clean \
+		lint format check-format vet docker-vet \
+		build-linux docker \
+		unit-test unit-test-race build-docker-test docker-func-test \
+		build-metrics docker-metrics \
+		metrics-unit-test docker-metrics-test
+
+# VERSION is the source revision that executables and images are built from.
+VERSION ?= $(shell git describe --tags --always --dirty || echo "unknown")
+
+# DESTDIR is where distribution output (container images) is placed.
+DESTDIR ?= .
+
+# IMAGE is the primary AWS VPC CNI plugin container image.
+IMAGE ?= amazon/amazon-k8s-cni
+IMAGE_NAME ?= $(IMAGE)$(IMAGE_ARCH_SUFFIX):$(VERSION)
+IMAGE_DIST = $(DESTDIR)/$(subst /,_,$(IMAGE_NAME)).tar.gz
+# METRICS_IMAGE is the CNI metrics publisher sidecar container image.
+METRICS_IMAGE ?= amazon/cni-metrics-helper
+METRICS_IMAGE_NAME ?= $(METRICS_IMAGE)$(IMAGE_ARCH_SUFFIX):$(VERSION)
+METRICS_IMAGE_DIST = $(DESTDIR)/$(subst /,_,$(METRICS_IMAGE_NAME)).tar.gz
+# TEST_IMAGE is the testing environment container image.
+TEST_IMAGE = amazon-k8s-cni-test
+TEST_IMAGE_NAME = $(TEST_IMAGE)$(IMAGE_ARCH_SUFFIX):$(VERSION)
+# These values derive ARCH and DOCKER_ARCH which are needed by dependencies in
+# image build defaulting to system's architecture when not specified.
+#
+# UNAME_ARCH is the runtime architecture of the building host.
+UNAME_ARCH = $(shell uname -m)
+# ARCH is the target architecture which is being built for.
+#
+# These are pairs of input_arch to derived_arch separated by colons:
+ARCH ?= $(lastword $(subst :, ,$(filter $(UNAME_ARCH):%,x86_64:amd64 aarch64:arm64)))
+# DOCKER_ARCH is the docker specific architecture specifier used for building on
+# multiarch container images.
+DOCKER_ARCH ?= $(lastword $(subst :, ,$(filter $(ARCH):%,amd64:amd64 arm64:arm64v8)))
+# IMAGE_ARCH_SUFFIX is the `-arch` suffix included in the container image name.
+#
+# This is only applied to the arm64 container image by default. Override to
+# provide an alternate suffix or to omit.
+IMAGE_ARCH_SUFFIX ?= $(addprefix -,$(filter $(ARCH),arm64))
+# GOLANG_IMAGE is the building golang container image used.
+GOLANG_IMAGE ?= golang:1.13-stretch
+# For the requseted build, these are the set of Go specific build environment
+# variables.
+export GOARCH = $(ARCH)
+export GOOS = linux
+export CGO_ENABLED = 0
+# NOTE: Provided for local toolchains that require explicit module feature flag.
+export GO111MODULE = on
+export GOPROXY ?= direct
+
+# LDFLAGS is the set of flags used when building golang executables.
 LDFLAGS ?= -X main.version=$(VERSION)
+# ALLPKGS is the set of packages provided in source.
+ALLPKGS = $(shell go list ./...)
+# BINS is the set of built command executables.
+BINS = aws-k8s-agent aws-cni grpc-health-probe
```

Comment:
Nit: Should the `cni-metrics-helper` be here as well, or just on a separate line in the `clean:` target?

Reply:
Yep, it should be here. Slipped my mind, will add it to this list.